images/chromium-headful/client/src/app.vue

```diff
@@ -251,6 +255,47 @@
       }
     }
 
+    // KERNEL: begin custom resolution, frame rate, and readOnly control via query params
+
+    // Add a watcher so that when we are connected we can set the resolution from query params
+    @Watch('connected', { immediate: true })
+    onConnected(value: boolean) {
+      if (value) {
+        this.applyQueryResolution()
+      }
+    }
+
+    // Read ?width=, ?height=, and optional ?rate= (or their short aliases w/h/r) from the URL
+    // and set the resolution accordingly. If the current user is an admin we also request the
+    // server to switch to that resolution.
+    private applyQueryResolution() {
+      const params = new URL(location.href).searchParams
+
+      const width = parseInt(params.get('width') || params.get('w') || '')
+      const height = parseInt(params.get('height') || params.get('h') || '')
+      const rate = parseInt(params.get('rate') || params.get('r') || '30')
+
+      if (!isNaN(width) && !isNaN(height)) {
+        const resolution = { width, height, rate: isNaN(rate) ? 30 : rate }
+        this.$accessor.video.setResolution(resolution)
+        if (this.$accessor.user && this.$accessor.user.admin) {
+          this.$accessor.video.screenSet(resolution)
+        }
+      }
```

Comment:
flyby question: what's `server` mean in this context? we aren't actually changing os level resolution are we? mainly interested in understanding the interaction between this and replays!